lnd.go

```diff
@@ -667,25 +667,54 @@ func Main(cfg *Config, lisCfg ListenerCfg, implCfg *ImplementationCfg,
 	ltndLog.Infof("Waiting for chain backend to finish sync, "+
 		"start_height=%v", bestHeight)
 
+	type syncResult struct {
+		synced        bool
+		bestBlockTime int64
+		err           error
+	}
+
+	var syncedResChan = make(chan syncResult, 1)
+
 	for {
-		if !interceptor.Alive() {
+		// We check if the wallet is synced in a separate goroutine as
+		// the call is blocking, and we want to be able to interrupt it
+		// if the daemon is shutting down.
+		go func() {
+			synced, bestBlockTime, err := activeChainControl.Wallet.
+				IsSynced()
+			syncedResChan <- syncResult{synced, bestBlockTime, err}
+		}()
+
+		select {
+		case <-interceptor.ShutdownChannel():
```

Comment:
wonder why there exist the ShutdownChannel and the quit channel, can they be summarised ?

Reply:
Not sure what you mean. There is no `quit` channel in `lnd.go`, only the interceptor's shutdown channel (which fires when the user presses Ctrl+C or the daemon receives the interrup/kill signal).

Reply:
prior we were listening to `if !interceptor.Alive() {` which listens to the quit channel of the interceptor, now we listen to the `shutdownchannel` of the interceptor, wondering if that makes a difference.

Reply:
Ah, I see what you mean. I don't think it makes a difference. The `Alive()` method is just a way to get a boolean instead of needing to select on a channel. But when a shutdown is triggered, both the shutdown channel and the quit channel fire.

Reply:
Just for reference, I also tested locally before updating with the removal of the `if !interceptor.Alive() {` check, that doing so made no difference.